man/systemd-run.xml

```diff
@@ -536,6 +536,30 @@
     the launched process.</para>
   </refsect1>
 
+  <refsect1>
+    <title>Environment</title>
```

Comment:
we usually don't document these kind of workaround-ish env vars in manpages, but exclusively in docs/ENVIRONMENT.md (it has tools-specific sections too)

Reply:
I don't think it's workaround-ish. It's a very user-facing feature, so it's on the same level as SYSTEMD_EDITOR that systemctl(1) documents.

Reply:
I don't mind having it in the manpage too, but please also document it in the markdown doc, as we want to list all variables there

Reply:
I updated the doc as well.

Reply:
Let's not document these in man pages. We use envvar exactly because we don't want to claim the same level of support as options

Reply:
> Let's not document these in man pages. We use envvar exactly because we don't want to claim the same level of support as options

You _just_ said minutes ago you wanted an envvar because it didn't look very useful as an option, not because you didn't want to support the feature.

I've removed that commit for now, but…

I think "claim the level of support" is kind of the root of the problem, though. There's a difference between not supporting options that are genuinely edge cases in terms of code, and not supporting options that literally just turn off in-your-face UI frills. I would understand not supporting an option that _adds_ the "set title" feature – but not an option that _removes_ it after previous versions were not-doing the same thing without any issues.

(In fact, how come setting a custom background is supported, and there is a `--background=` option for that, but _not_ setting a background is not-supported and relegated to /usr/share/doc? Is there a code reason, or a test matrix reason, or a maintenance reason that makes them different? How is either of those any different from not setting the terminal title?)

Frankly, speaking as someone who's been around for 13 years, and who used to run systemd straight off Git back when the versions were two-digit, just so I could get to play with the newest stuff... all this stuff has _completely_ drained my patience over the years. I no longer want to build a fresh systemd to see what's in it, or to set up Secure Boot and TPMs the way I used to, and I absolutely do not want tools to be 'in my face' anymore. I don't want fancy backgrounds, or fancy emojis, or anything more than a bit of color here and there – I just want my system tools to stay out of the way as much as they can.

The sheer amount of _stuff_ that's being added is absolutely overwhelming. It's _exhausting._  There are things that don't really bother me other than "well, I guess they're in my /usr/lib now, that's fine", but when it gets to having good, useful tools suddenly change in behavior like this, without as much as an option to undo that, it's frustrating to say the least.

So the least I can do, I think, is to try to get things documented – in the standard place people look at.

Reply:
i think we might want to introduce a new man page for terminal/UI-related env vars, and then we should document SYSTEMD_URLIFY=, SYSTEMD_COLORS=, SYSTEMD_EDITOR= and so on there, and maybe also these new env vars. Right now we embed the same list of env vars over and over again into quite a bunch of man pages, even though it's always the exact same text and not really the primary function of the relevant tools, hence a bit noisy. Have a single, good (and then a bit more verbose) man page for also those options that tweak the user experience sounds like a good approach to me.

Reply:
> i think we might want to introduce a new man page for terminal/UI-related env vars, and then we should document SYSTEMD_URLIFY=, SYSTEMD_COLORS=, SYSTEMD_EDITOR= and so on there, and maybe also these new env vars. Right now we embed the same list of env vars over and over again into quite a bunch of man pages, even though it's always the exact same text and not really the primary function of the relevant tools, hence a bit noisy. Have a single, good (and then a bit more verbose) man page for also those options that tweak the user experience sounds like a good approach to me.

Previous rant aside, I'd really like to have *all* envvars documented as manual pages, regardless of how much they are supported.

But I don't really agree that it's noise to embed it in the tools' manpages; it's very useful to have such information directly where it is relevant. And I can say it's really not fun to chase docs across long lists of `SEE ALSO` references, such as "is x-foo documented or fstab(5) or in systemd.mount(5) or in systemd-fstab-generator(8)?" that I keep forgetting every time.

So if the manpage size is a concern, I'd at least embed a terse paragraph-form list such as "The common environment variables SYSTEMD_COLORS, SYSTEMD_URLIFY, [etc] are supported by this tool; see systemd.envvars(π) for their description" rather than outright removing them all.

Reply:
> So if the manpage size is a concern, I'd at least embed a terse paragraph-form list such as "The common environment variables SYSTEMD_COLORS, SYSTEMD_URLIFY, [etc] are supported by this tool; see systemd.envvars(π) for their description" rather than outright removing them all.

that's fine by me